meta.go

```diff
@@ -810,124 +810,131 @@ func (c marshaledContainer) MarshalYAML() (interface{}, error) {
 			// to use the otherwise-reserved juju-* namespace.
-			if !meta.Subordinate || role != RoleRequirer || rel.Scope != ScopeContainer {
+			if !m.Subordinate || role != RoleRequirer || rel.Scope != ScopeContainer {
 				if reserved, _ := reservedName(name); reserved {
-					return fmt.Errorf("charm %q using a reserved relation name: %q", meta.Name, name)
+					return errors.Errorf("charm %q using a reserved relation name: %q", m.Name, name)
 				}
 			}
 			if role != RoleRequirer {
 				if reserved, _ := reservedName(rel.Interface); reserved {
-					return fmt.Errorf("charm %q relation %q using a reserved interface: %q", meta.Name, name, rel.Interface)
+					return errors.Errorf("charm %q relation %q using a reserved interface: %q", m.Name, name, rel.Interface)
 				}
 			}
 			if names[name] {
-				return fmt.Errorf("charm %q using a duplicated relation name: %q", meta.Name, name)
+				return errors.Errorf("charm %q using a duplicated relation name: %q", m.Name, name)
 			}
 			names[name] = true
 		}
 		return nil
 	}
-	if err := checkRelations(meta.Provides, RoleProvider); err != nil {
+	if err := checkRelations(m.Provides, RoleProvider); err != nil {
 		return err
 	}
-	if err := checkRelations(meta.Requires, RoleRequirer); err != nil {
+	if err := checkRelations(m.Requires, RoleRequirer); err != nil {
 		return err
 	}
-	if err := checkRelations(meta.Peers, RolePeer); err != nil {
+	if err := checkRelations(m.Peers, RolePeer); err != nil {
 		return err
 	}
 
-	if err := validateMetaExtraBindings(meta); err != nil {
-		return fmt.Errorf("charm %q has invalid extra bindings: %v", meta.Name, err)
+	if err := validateMetaExtraBindings(m); err != nil {
+		return errors.Errorf("charm %q has invalid extra bindings: %v", m.Name, err)
 	}
 
 	// Subordinate charms must have at least one relation that
 	// has container scope, otherwise they can't relate to the
 	// principal.
-	if meta.Subordinate {
+	if m.Subordinate {
 		valid := false
-		if meta.Requires != nil {
-			for _, relationData := range meta.Requires {
+		if m.Requires != nil {
+			for _, relationData := range m.Requires {
 				if relationData.Scope == ScopeContainer {
 					valid = true
 					break
 				}
 			}
 		}
 		if !valid {
-			return fmt.Errorf("subordinate charm %q lacks \"requires\" relation with container scope", meta.Name)
+			return errors.Errorf("subordinate charm %q lacks \"requires\" relation with container scope", m.Name)
 		}
 	}
 
-	for _, series := range meta.Series {
-		if !IsValidSeries(series) {
-			return fmt.Errorf("charm %q declares invalid series: %q", meta.Name, series)
+	if m.Format() == FormatV1 {
+		for _, series := range m.Series {
+			if !IsValidSeries(series) {
+				return errors.Errorf("charm %q declares invalid series: %q", m.Name, series)
+			}
+		}
+	} else {
+		// Version 2 of the metadata should not delcare a series.
+		if len(m.Series) > 0 {
+			return errors.Errorf("charm %q declares both series and systems", m.Name)
```

Comment:
I know upstream has not yet changed, but this will be "bases".

Reply:
Added a TODO.